tests/unit/adapters/application-test.js

```diff
@@ -29,103 +29,187 @@ moduleFor('adapter:application', 'Unit | Adapter | application', {
 });
 
 test('#find calls #findOne when options arg is a string', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject();
   sandbox.stub(adapter, 'findOne', function () { return RSVP.Promise.resolve(null); });
   let promise = adapter.find('1');
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.findOne.calledOnce, 'findOne called');
-  assert.ok(adapter.findOne.calledWith('1'), 'findOne called with "1"');
+  promise.then(() => {
+    assert.ok(adapter.findOne.calledOnce, 'findOne called');
+    assert.ok(adapter.findOne.calledWith('1'), 'findOne called with "1"');
+    done();
+  });
 });
 
 test('#find calls #findOne when options arg is an object having an id property', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject();
   sandbox.stub(adapter, 'findOne', function () { return RSVP.Promise.resolve(null); });
-  let options = { id: '1', query: {sort: '-date'} };
+  let options = {id: '1', query: {sort: '-date'}};
   let promise = adapter.find(options);
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.findOne.calledOnce, 'findOne called');
-  assert.ok(adapter.findOne.calledWith(options.id, options.query), 'findOne called with `"1"` and query `{"sort": "-date"}`');
+  promise.then(() => {
+    assert.ok(adapter.findOne.calledOnce, 'findOne called');
+    assert.ok(
+      adapter.findOne.calledWith(options.id, options.query),
+      'findOne called with `"1"` and query `{"sort": "-date"}`'
+    );
+    done();
+  });
 });
 
+// Yeah... findQuery is used when options arg is not a string or an object.
+// undefined is not a string or an object, but so is a number.
+// TODO: id.toString() in to public methods? Or make this test more specific?
```

Comment:
Adapter#find is doesn't find anything it just branches between `findOne` or `findQuery`. See https://github.com/pixelhandler/ember-jsonapi-resources/blob/master/addon/adapters/application.js#L41-L90

If `findQuery` is called with a falsey argument the `options` parameter defaults to an empty object.

A resource identifier object as defined by the JSON API spec has an `id` which much be a string. see http://jsonapi.org/format/#document-resource-object-identification

I started with the premise the this library should stick close to the spec, so using a number seems questionable. I'm concerned that accommodating a number as an `id` and casting to a `String` could be a step away from sticking closely to the spec. So `findOne` should only receive a string.

I can't recall if I added any guard to the `id` property in the Resource prototype.

Maybe an assertion somewhere that warns when using a number as an `id` would be good (instead of casting to a string). What do you think?